fili-core/src/test/groovy/com/yahoo/bard/webservice/web/responseprocessors/DruidPartialDataResponseProcessorSpec.groovy

```diff
@@ -0,0 +1,258 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web.responseprocessors
+
+import com.yahoo.bard.webservice.druid.client.HttpErrorCallback
+import com.yahoo.bard.webservice.druid.model.datasource.DataSource
+import com.yahoo.bard.webservice.druid.model.query.DruidAggregationQuery
+import com.yahoo.bard.webservice.druid.model.query.QueryContext
+import com.yahoo.bard.webservice.table.ConstrainedTable
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.node.ArrayNode
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module
+
+import org.joda.time.Interval
+
+import spock.lang.Specification
+
+class DruidPartialDataResponseProcessorSpec extends Specification {
+    private static final ObjectMapper MAPPER = new ObjectMapper()
+            .registerModule(new Jdk8Module().configureAbsentsAsNulls(false))
+
+    ResponseProcessor next
+    HttpErrorCallback httpErrorCallback
+    DruidAggregationQuery druidAggregationQuery
+    DruidPartialDataResponseProcessor druidPartialDataResponseProcessor
+
+    def setup() {
+        next = Mock(ResponseProcessor)
+        httpErrorCallback = Mock(HttpErrorCallback)
+        druidAggregationQuery = Mock(DruidAggregationQuery)
+        next.getErrorCallback(druidAggregationQuery) >> httpErrorCallback
+        druidPartialDataResponseProcessor = new DruidPartialDataResponseProcessor(next)
+    }
+
+    def "getOverlap returns intersection between Druid intervals and Fili intervals in case of #caseDescription"() {
+        given:
+        JsonNode json = MAPPER.readTree(jsonInString)
+
+        DataSource dataSource = Mock(DataSource)
+        ConstrainedTable constrainedTable = Mock(ConstrainedTable)
+
+        constrainedTable.getAvailableIntervals() >> new SimplifiedIntervalList(
+                availableIntervals.collect{it -> new Interval(it)}
+        )
+        dataSource.getPhysicalTable() >> constrainedTable
+        druidAggregationQuery.getDataSource() >> dataSource
+
+        expect:
+        druidPartialDataResponseProcessor.getOverlap(json, druidAggregationQuery) == new SimplifiedIntervalList(
+                expected.collect{it -> new Interval(it)}
+        )
+
+        where:
+        jsonInString | availableIntervals | expected | caseDescription
+        '''
+        {
+            "response": [{"k1":"v1"}],
+            "X-Druid-Response-Context": {
+                "uncoveredIntervals": [
+                    "2016-11-22T00:00:00.000Z/2016-12-18T00:00:00.000Z",
+                    "2016-12-25T00:00:00.000Z/2017-01-03T00:00:00.000Z"
+                ],
+                "uncoveredIntervalsOverflowed": false
+            },
+            "status-code": 200
```

Comment:
It looks like your Json instring is the same across all the test cases. If so, lift it up out of the where clause and put it into the test.